kubeadm/v1.15.0/KubeClusterHelper.psm1

```diff
@@ -0,0 +1,1360 @@
+                    m_serviceStatus.dwCurrentState = ServiceState.SERVICE_STOPPED;
+                }
+                finally
+                {
+                    SetServiceStatus(ServiceHandle, ref m_serviceStatus);
+                    if (m_writer != null) m_writer.Flush();
+                }
+            }
+
+            protected override void OnStop() {
+                Console.WriteLine("OnStop $ServiceName");
+                try 
+                {
+                    m_serviceStatus.dwCurrentState = ServiceState.SERVICE_STOPPED;
+                    if (m_process != null)
+                    {
+                        m_process.Kill();
+                        m_process.WaitForExit();
+                        m_process.Close();
+                        m_process.Dispose();
+                        m_process = null;
+                    }
+                    Console.WriteLine("OnStop - Successfully stopped the service ");
+                } 
+                catch (Exception e)
+                {
+                    Console.WriteLine("OnStop - Failed to stop the service : " + e.Message);
+                    m_serviceStatus.dwCurrentState = ServiceState.SERVICE_RUNNING;
+                }
+                finally
+                {
+                    SetServiceStatus(ServiceHandle, ref m_serviceStatus);
+                    if (m_writer != null) m_writer.Flush();
+                }
+            }
+
+            public static void Main() {
+                System.ServiceProcess.ServiceBase.Run(new ScmService_$ServiceName());
+            }
+        }
+"@
+
+    Add-Type -TypeDefinition $svcSource -Language CSharp `
+        -OutputAssembly $SvcBinary -OutputType ConsoleApplication   `
+        -ReferencedAssemblies "System.ServiceProcess" -Debug:$false
+
+    return $SvcBinary
+}
+
+function RemoveService()
+{
+    param
+    (
+        [parameter(Mandatory=$true)] [string] $ServiceName
+    )
+    $src = Get-Service -Name $ServiceName  -ErrorAction SilentlyContinue
+    if ($src) {
+        Stop-Service $src
+        sc.exe delete $src;
+    }
+}
+
+function InstallDockerD()
+{
+    Param(
+    [ValidateSet("docker")] [parameter(Mandatory = $false)] $Version = "docker",
+    $DestinationPath
+    ) 
+    # Add path to this PowerShell session immediately
+    $env:path += ";$env:ProgramFiles\Docker"
+    # For persistent use after a reboot
+    $existingMachinePath = [Environment]::GetEnvironmentVariable("Path",[System.EnvironmentVariableTarget]::Machine)
+    [Environment]::SetEnvironmentVariable("Path", $existingMachinePath + ";$env:ProgramFiles\Docker", [EnvironmentVariableTarget]::Machine)
+
+    $cmd = get-command docker.exe -ErrorAction SilentlyContinue
+    if (!$cmd)
+    {
+        $dockerVersion = $Version
+        DownloadFile  "https://master.dockerproject.org/windows/x86_64/${dockerVersion}.zip" -Destination "$env:TEMP\$dockerVersion.zip" 
```

Comment:
This Docker version is very recent which is nice, but it fails the kubeadm pre-flight check for me :(

Reply:
```Install-Module -Name DockerMsftProvider -Repository PSGallery -Force```

```Install-Package -Name docker -ProviderName DockerMsftProvider -Force -RequiredVersion 18.09  ```

Can you try updating the code with the above and see if that works?

Reply:
yes that works. I uninstalled Docker & re-validated to make sure.